File: selection_sort/selection_sort.py


def findSmallest(arr):
    smallest = arr[0]
    smallest_index = 0

    for i in range(1, len(arr)):
        if arr[i] < smallest:
            smallest = arr[i]
            smallest_index = i
    
    return smallest_index


def selectionSort(arr):
    newArr = []

    for i in range(len(arr)):
        smallest = findSmallest(arr)
        newArr.append(arr.pop(smallest))
    
    return newArr

arr = [5, 3, 6, 2, 10]


def newSelectionSort(arr):

    for i in range(len(arr)):
        smallest_index = i
        for j in range(i + 1, len(arr)):
            if arr[smallest_index] > arr[j]:
                smallest_index = j
        arr[i], arr[smallest_index] = arr[smallest_index], arr[i]

    return arr


print(newSelectionSort(arr))

File: selection_sort/selection_sort.js


function selectionSort(arr) {
    
    for (let i = 0; i < arr.length - 1; i++) {
        
        let smallest = i
        for (let j = i + 1; j < arr.length; j++) {
            if (arr[j] < arr[smallest]) {
                smallest = j
            }
        }
        
        [arr[i], arr[smallest]] = [arr[smallest], arr[i]]
    }

    return arr
}

const arr = [5, 3, 6, 2, 10]


console.log(selectionSort(arr))